### PR TITLE
MUIC-269: [iOS] User wants to see chat history without starting an engagement

### PR DESCRIPTION
**Pull request by kuldar-daniel**

### CODE REVIEW

File: GliaWidgets/ViewModel/Chat/ChatViewModel.swift [hunk 1]
```diff
@@ -198,6 +217,28 @@ class ChatViewModel: EngagementViewModel, ViewModel {
             case .stopped:
                 engagementAction?(.showEndButton)
             }
+            
+            pendingMessages.forEach({ [weak self] outgoingMessage in
+                self?.interactor.send(outgoingMessage.content, attachment: nil) { message in
```
Comment (dukhovnyi) [inline GliaWidgets/ViewModel/Chat/ChatViewModel.swift:222]:
```suggestion
                self?.interactor.send(outgoingMessage.content, attachment: nil) { [weak self] message in
```

File: GliaWidgets/ViewModel/Chat/ChatViewModel.swift [hunk 2]
```diff
@@ -198,6 +217,28 @@ class ChatViewModel: EngagementViewModel, ViewModel {
...
+                    
+                    self.replace(
+                        outgoingMessage,
+                        uploads: [],
+                        with: message,
+                        in: self.messagesSection
+                    )
+
+                    self.action?(.scrollToBottom(animated: true))
+                } failure: { _ in
```
Comment (dukhovnyi) [inline GliaWidgets/ViewModel/Chat/ChatViewModel.swift:233]:
```suggestion
                } failure: { [weak self] _ in
```

File: GliaWidgets/Coordinator/RootCoordinator.swift [hunk 3]
```diff
@@ -136,15 +135,22 @@ class RootCoordinator: SubFlowCoordinator, FlowCoordinator {
             case .back:
                 switch self?.engagement {
                 case .chat:
-                    self?.gliaViewController?.minimize(animated: true)
+                    if !(self?.interactor.isEngaged ?? false) {
+                        self?.popCoordinator()
+                        self?.end()
+                    } else {
+                        self?.gliaViewController?.minimize(animated: true)
+                    }
```
Comment (dukhovnyi) [inline GliaWidgets/Coordinator/RootCoordinator.swift:143]: imho, it is easier to read code when it doesn't contain additional negation `!`. So, I'd like to rework it a little bit
```swift
if self?.interactor.isEngaged == true {
    self?.gliaViewController?.minimize(animated: true)
} else {
    self?.popCoordinator()
    self?.end()
}
```

File: GliaWidgets/ViewModel/Chat/ChatViewModel.swift [hunk 4]
```diff
@@ -334,27 +375,54 @@ extension ChatViewModel {
...
+        if interactor.isEngaged {
+            let item = ChatItem(with: outgoingMessage)
+            appendItem(item, to: messagesSection, animated: true)
+            
+            uploader.removeAllUploads()
+            action?(.removeAllUploads)
+            action?(.scrollToBottom(animated: true))
+            let messageTextTemp = messageText
+
+            interactor.send(messageTextTemp, attachment: attachment) { message in
```
Comment (dukhovnyi) [inline GliaWidgets/ViewModel/Chat/ChatViewModel.swift:388]: Weak reference is required here, let's add `[weak self]` here.
```suggestion
            interactor.send(messageTextTemp, attachment: attachment) { [weak self] message in
```

File: GliaWidgets/ViewModel/Chat/ChatViewModel.swift [hunk 5]
```diff
@@ -334,27 +375,54 @@ extension ChatViewModel {
...
+                )
 
-        interactor.send(messageTextTemp, attachment: attachment) { message in
-            self.replace(
-                outgoingMessage,
-                uploads: uploads,
-                with: message,
-                in: self.messagesSection
+                self.action?(.scrollToBottom(animated: true))
+            } failure: { _ in
```
Comment (dukhovnyi) [inline GliaWidgets/ViewModel/Chat/ChatViewModel.swift:397]: Weak reference is required here, let's add `[weak self]` here.
```suggestion
            } failure: { [weak self] _ in
```

File: GliaWidgets/Coordinator/RootCoordinator.swift (hunk 3 above)
Comment (kuldar-daniel) [reply GliaWidgets/Coordinator/RootCoordinator.swift]: Good catch. Had to refactor it anyway for bubble logic, should be good now

**Review by dukhovnyi — APPROVED** [review]

**Comment by dukhovnyi** [comment]
👍

**Review by VladKupriienko — APPROVED** [review]

**Pull request merged**